clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -135,7 +138,22 @@ public VeniceChangelogConsumerImpl(
       ChangelogClientConfig changelogClientConfig,
       PubSubConsumerAdapter pubSubConsumer) {
     this.pubSubConsumer = pubSubConsumer;
-    this.storeName = changelogClientConfig.getStoreName();
+
+    // TODO: putting the change capture case here is a little bit weird. The view abstraction should probably
+    // accommodate
+    // this case, but it doesn't seem that clean in this case. Change capture topics don't behave like view topics as
+    // they only contain nearline data whereas views are full version topics. So for now it seems legit to have this
+    // caveat, but we should perhaps think through this in the future if we think we'll have more nearline data only
+    // cases for view topics. Change capture topic naming also doesn't follow the view_store_store_name_view_name
+    // naming pattern, making tracking it even weirder.
+    if (changelogClientConfig.getViewName() == null || changelogClientConfig.getViewName().isEmpty()
+        || changelogClientConfig.isBeforeImageView()) {
+      this.storeName = changelogClientConfig.getStoreName();
+    } else {
+      this.storeName =
+          ChangeCaptureView.getViewStoreName(changelogClientConfig.getStoreName(), changelogClientConfig.getViewName());
```

Comment:
Does it matter that `localStateTopicNameTemp` and `localStateTopicName` in `InternalLocalBootstrappingVeniceChangelogConsumer` will still use the corresponding store name `changelogClientConfig.getStoreName()` even when view is specified?

Reply:
yeah.... admittedly it looks sketchy, but, it's just for naming of filepaths.  I don't see an obvious bug excluding it, HOWEVER, there is potential issue if you ever had local state and you changed your view source.  In that case, we should probably pad the view name if only to make certain that state doesn't drag over between runs.

Reply:
Do you foresee users using CC for both the store and its views in the same instance/host? The same problem could apply right?